src/Core/src/Platform/Gtk/NavigationView.cs

```diff
@@ -1,11 +1,43 @@
+using System;
+using System.Linq;
+using Gtk;
+
 namespace Microsoft.Maui.Platform
 {
 
 	public class NavigationView : Gtk.Box
 	{
-
+		Gtk.Widget? pageWidget = null;
+		IMauiContext? mauiContext = null;
+		
 		public NavigationView() : base(Gtk.Orientation.Horizontal, 0) { }
 
+		public void Connect(IStackNavigationView virtualView)
+		{
+			mauiContext = virtualView.Handler?.MauiContext;
+		}
+
+		public void Disconnect(IStackNavigationView virtualView)
+		{
+		}
+
+		public void RequestNavigation(NavigationRequest request)
+		{
+			var page = request.NavigationStack.Last(); // stack top is last
+			var newPageWidget = page.ToPlatform(mauiContext!);
+			if (pageWidget == null)
```

Comment:
@webwarrior-ws @parhamsaremi should use `is`?

Reply:
Done